app.rb

```diff
@@ -0,0 +1,347 @@
+require_relative 'helper'
+
+def load_books
+  @books = PreserveBooks.new.gets_books || []
+end
+
+def load_music_albums
+  @music_albums = PreserveMusicAlbums.new.gets_music_albums || []
+end
+
+def load_movies
+  @movies = PreserveMovies.new.gets_movies || []
+end
+
+def load_games
+  @games = PreserveGames.new.gets_games || []
+end
+
+def load_genres
+  @genres = PreserveGenres.new.gets_genres || []
+end
+
+def load_labels
+  @labels = PreserveLabels.new.gets_labels || []
+end
+
+def load_authors
+  @authors = PreserveAuthors.new.gets_authors || []
+end
+
+def load_sources
+  @sources = PreserveSources.new.gets_sources || []
+end
+
+def add_genre
+  print 'Genre: '
+  genre_name = gets.chomp.to_s
+  @genre = Genre.new(genre_name)
+  return if @genres.any? { |genre| genre.genre_name == genre_name }
+
+  @genres << @genre
+end
+
+def add_author(item)
+  if item == 'music_album'
+    print 'Artist: '
+    first_name = gets.chomp.to_s
+    @author = Author.new(first_name, '')
+  elsif item == 'movie'
+    print "Director's First Name: "
+    first_name = gets.chomp.to_s
+
+    print "Director's Last Name: "
+    last_name = gets.chomp.to_s
+
+    @author = Author.new(first_name, last_name)
+  else
+    print 'Author First Name: '
+    first_name = gets.chomp.to_s
+    print 'Author Last Name: '
+    last_name = gets.chomp.to_s
+    @author = Author.new(first_name, last_name)
+  end
+
+  return if @authors.any? { |author| author.first_name + author.last_name == first_name.to_s + last_name.to_s }
+
+  @authors << @author
+end
+
+def add_source
+  print 'Source: '
+  source_name = gets.chomp.to_s
+  @source = Source.new(source_name)
+
+  return if @sources.any? { |source| source.source_name == source_name }
+
+  @sources << @source
+end
+
+def add_label
+  print 'Title: '
+  title = gets.chomp.to_s
+  print 'Color: '
+  color = gets.chomp.to_s
+
+  @label = Label.new(title, color)
+
+  return if @labels.any? { |label| label.title + label.color == title + color }
+
+  @labels << @label
+end
+
+def add_item_properties(item)
+  add_genre
+  add_author(item)
+  add_source
+  add_label
+
+  print 'Publish Date (YEAR): '
+  @publish_date = gets.chomp
+
+  if @publish_date.match?(/\A\d+\z/)
+    @publish_date = @publish_date.to_i
+
+  else
+    puts "ERROR: Invalid answer. Value set to 'Unkown'"
+    @publish_date = 0
+  end
+
+  @item_params = { genre: @genre, author: @author, source: @source,
+                   label: @label, publish_date: @publish_date }
+end
+
+def list_all_books
+  if @books.empty?
+    puts "\nNo books yet"
```

Comment:
- [optional] please instead of just saying no books yet or no music yet when checking the list, we can be more polite by using friendly language in the app, which can significantly enhance the user experience and make users feel more comfortable and engaged. E.g. "sorry, you haven't added any book yet"/"Unfortunately, there are no books in our list yet"